okhttp/src/main/kotlin/okhttp3/internal/tls/TrustManagerAndroid.kt

```
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.tls
+
+import android.os.Build
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+import java.security.cert.X509Certificate
+import javax.net.ssl.X509TrustManager
+import okhttp3.internal.SuppressSignatureCheck
+
+internal open class TrustManagerOverrideAndroid(
+  val predicate: (String) -> Boolean,
+  val trustManager: AndroidTrustManager
+)
+
+internal class TrustManagerWrapperAndroid(val trustManager: X509TrustManager) : AndroidTrustManager {
+  val delegateMethod by lazy { lookupAndroidDelegateMethod() }
```

should this be non-lazy to fail fast?